apm-sniffer/apm-sdk-plugin/mongodb-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mongodb/v3/support/MongoInfoHelper.java

```diff
@@ -143,4 +150,66 @@ private static String limitFilter(String filter) {
         }
     }
 
+    /**
+     * Get the databaseName
+     *
+     * @param obj operation
+     * @return databaseName
+     */
+    @SuppressWarnings("rawtypes")
+    public static String getMongoDatabaseName(Object obj) throws NoSuchFieldException, IllegalAccessException {
+        String databaseName = "";
+        if (obj instanceof CountOperation) {
+            Field namespace = CountOperation.class.getDeclaredField(NAMESPACE_FIELD_NAME);
+            namespace.setAccessible(true);
+            databaseName = getOrDefaultDatabaseName((MongoNamespace) namespace.get(obj));
+        } else if (obj instanceof DistinctOperation) {
+            Field namespace = DistinctOperation.class.getDeclaredField(NAMESPACE_FIELD_NAME);
+            namespace.setAccessible(true);
+            databaseName = getOrDefaultDatabaseName((MongoNamespace) namespace.get(obj));
+        } else if (obj instanceof MapReduceWithInlineResultsOperation) {
+            databaseName = getOrDefaultDatabaseName(((MapReduceWithInlineResultsOperation) obj).getNamespace());
+        } else if (obj instanceof DeleteOperation) {
```

Comment:
`DeleteOperation `、`InsertOperation `、`UpdateOperation `，It seems that it can be simplified to `BaseWriteOperation`

Reply:
`reflection` is also slow from a performance perspective. You could run a benchmark to see the result.